server/game/cards/13.6-LMHR/TheQueensRetinue.js

```diff
@@ -7,12 +7,14 @@ class TheQueensRetinue extends DrawCard {
             location: 'shadows',
             handler: context => {
                 context.player.putIntoPlay(this);
-                for(let opponent of this.game.getOpponents(this.controller)) {
-                    if(opponent.canDraw()) {
-                        opponent.drawCardsToHand(2);
+                if(this.location === 'play area') {

```

Comment:
Thanks for picking this up so quick!

This would work right now. In the near term future, every game action will be asynchronous. In this instance, `putIntoPlay` would generate on `onCardEntersPlay` event and the location wouldn't be set to 'play area' until after the interrupts for that event have been resolved. So putting into play and immediately doing an `if` would fail.

The better check would be to prevent the ability from triggering when the card can't be put into play. This could be done by adding a `condition` to the ability that checks that `this` can be put into play. Check the other similar "put into play from shadows for some other cost" cards.

Reply:
Thank you for your feedback. I also checked the other 4 for 4 bicon shadow (4) cards every faction got and added/fixed the condition on them.